qiskit/transpiler/passes/utils/gates_basis.py

```diff
@@ -13,27 +13,51 @@
 """Check if all gates in the DAGCircuit are in the specified basis gates."""
 
 from qiskit.transpiler.basepasses import AnalysisPass
+from qiskit.transpiler.exceptions import TranspilerError
 
 
 class GatesInBasis(AnalysisPass):
     """Check if all gates in a DAG are in a given set of gates"""
 
-    def __init__(self, basis_gates):
+    def __init__(self, basis_gates=None, target=None):
         """Initialize the GatesInBasis pass.
 
         Args:
             basis_gates (list): The list of strings representing the set of basis gates.
```

Comment:
Should these docstring types be annotated within Optional[T]?

Reply:
I wouldn't say it's a hard requirement, we're not actually using type annotations here so mypy wouldn't be able to resolve that it's optional anyway